Repositories data

JuanitoFatas opened the pull request:
<kbd>[View Rendered README.md](https://github.com/jollygoodcode/data.rubyasia.com/blob/c9f01fe/README.md)</kbd>

<kbd>[View Data](https://github.com/jollygoodcode/data.rubyasia.com/commit/c9f01fe)</kbd>

JuanitoFatas committed:
Update README with fetch repositories [ci skip]

JuanitoFatas committed:
Introduce Repo class to fetch repositories

Will fetch from 7 days since yesterday.
The file name format is repo-{date_started_at}_{date_ended_at}.json
Current criteria: language:Ruby stars:>20

winston commented on app/repo.rb:
```diff
@@ -0,0 +1,133 @@
+
+    attr_reader :raw_region, :available_regions, :noramlized_regions, :regions
+
+    def determin_by_user_specified_region(user_specified_region)
```
`determine`

winston commented on app/repo.rb:
```diff
@@ -0,0 +1,133 @@
+      if user_specified_region == "all".freeze
+        noramlized_regions
+      else
+        if noramlized_regions.include?(user_specified_region)
```
`normalized`

winston commented:
So the criteria is:

>   "language:Ruby stars:>20 pushed:#{period}"

right?

Ruby repo with stars > 20 and last commit in the last 7 days?

If so, the data looks great! Am thinking if we should extend the period though.

JuanitoFatas commented:
> Ruby repo with stars > 20 and last commit in the last 7 days?

yes. A week before yesterday.

JuanitoFatas committed:
Fix typos, rearrange result

there is no "watchers" count from Summary Representations.

https://developer.github.com/v3/#summary-representations

JuanitoFatas committed:
$ rake fetch_repos[all]

JuanitoFatas committed:
Update README [ci skip]
